contributing_guidelines.md

```diff
@@ -0,0 +1,191 @@
+
+[Top](#top)
+
+###<a name="section6.2"></a>Pull requests
+Any contribution MUST be done through a new opened pull request (PR). These PRs MUST compare certain branch at any forked repository against the `develop` base branch in the main repository.
+
+At least one people from Telefónica’s team MUST participate in the PR review, and MUST agree with the PR before merging it. The agreement MAY be expressed with the “LGTM” (looks good to me) keyword. In order to agilize the PR process, this kind of review MUST be exclusively related to the alignment of the PR with these contribution guidelines. Any comment regarding the content MAY be added anyway.
+
+When a PR regarding the `cygnus-common` folder is done, then the review MUST agree both with the guidelines fulfillment and the content.
+
+Internally to every team, private code reviews SHOULD be done before pull requesting to the main repository.
+
+[Top](#top)
+
+###<a name="section6.3"></a>Contribution contents
+Every contribution/PR MUST include:
+
+* The code implementing the feature/hardening/bug/task.
+* Unit tests.
+* Documentation.
+
+Other tests MAY be included (acceptance, e2e, performance).
+
+Every contribution/PR MUST also add a new line in a special file within the root of the main repository, `CHANGES_NEXT_RELEASE`. The format of each line MUST follow this format:
+
+    - [<agent_name>] [feature|hardening|bug|task] <short description> (#<issue number>)
+
+Where <i>short description</i> MAY enclose other “[...]” sublevels. For instance:
+
+    - [cygnus-ngsi] [hardening] [grouping rules] Precompile regexes (#209)
+
+[Top](#top)
+
+###<a name="section6.4"></a>Commits nad squashing
+Commits within PRs MUST include a comment following this format:
+
+    [<agent name>] [issue number] <short description>
+
+Where <i>short description</i> MAY enclose other “[...]” sublevels. For instance:
+
+    [cygnus-ngsi] [873] Update CHANGES_NEXT_RELEASE
+    [cygnus-ngsi] [873] Add support for pattern storage
+    [cygnus-ngsi] [873] Add regex compilation
+
+With regards to the squashing policy, the main repository MUST be configured with the <i>Allow Squash Merging</i> option.
+
+[Top](#top)
+
+##<a name="section7"></a>Releasing
+When generating a new version of Cygnus from the main repository, all the agents MUST be released at the same time as a whole.
+
+A minor version (0.X.0, at the moment of writing this the last release number is 0.13.0) of Cygnus MUST be released at the end of each sprint/milestone. A sprint SHOULD comprise a natural month, however sometimes the sprints MAY comprise a different period, for instance a month and a half or half a month (usually, in order to adapt to holydays time). Every sprint MUST be scheduled in advance by Telefónica’s team in the form of deadline in the related milestone. Agent teams SHOULD use this information in order to, internally, schedule the sprint in terms of issues to be implemented.
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/943/commits/7a7e87176ef7b38c0d193fa949b7e468094f8274